src/meta/processors/partsMan/CreateSpaceProcessor.cpp

```diff
@@ -119,6 +134,38 @@ CreateSpaceProcessor::pickHosts(PartitionID partId,
     return pickedHosts;
 }
 
+int32_t CreateSpaceProcessor::GetPartsNumbers() {
+    int32_t count = 0;
+    auto prefix = MetaServiceUtils::spacePrefix();
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto ret = kvstore_->prefix(kDefaultSpaceId, kDefaultPartId, prefix, &iter);
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
+        LOG(ERROR) << "List Spaces Failed";
+        return -1;
+    }
+
+    while (iter->valid()) {
+        auto spaceId = MetaServiceUtils::spaceId(iter->key());
+        auto partPrefix = MetaServiceUtils::partPrefix(spaceId);
+        auto spaceName = MetaServiceUtils::spaceName(iter->val());
+        std::unique_ptr<kvstore::KVIterator> partIter;
+        auto kvRet = kvstore_->prefix(kDefaultSpaceId, kDefaultPartId, partPrefix, &partIter);
+        if ( kvRet != kvstore::ResultCode::SUCCEEDED ) {
+            LOG(ERROR) << "List Partitions From " << spaceName
+                       << ", Space id " << spaceId << " Failed";
+            return -1;
+        }
+
+        while (partIter->valid()) {
```

Comment:
You can get the partition_num from space properties, don't need list parts.

Reply:
good point, Done